src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py

```diff
@@ -412,39 +420,46 @@ def execute(self):
 
 class ResourceGroupVCRTestBase(VCRTestBase):
     # pylint: disable=too-many-arguments
-    def __init__(self, test_file, test_name, run_live=False, debug=False, debug_vcr=False,
-                 skip_setup=False, skip_teardown=False):
+    def __init__(self, test_file, test_name, resource_group='vcr_resource_group', run_live=False,
+                 debug=False, debug_vcr=False, skip_setup=False, skip_teardown=False):
         super(ResourceGroupVCRTestBase, self).__init__(test_file, test_name, run_live=run_live,
-                                                       debug=debug, skip_setup=skip_setup,
+                                                       debug=debug, debug_vcr=debug_vcr,
+                                                       skip_setup=skip_setup,
                                                        skip_teardown=skip_teardown)
-        self.resource_group = 'vcr_resource_group'
+        self.resource_group_original = resource_group
+        random_tag = '_{}_'.format(''.join((choice(ascii_lowercase + digits) for _ in range(4))))
```

Comment:
According to https://docs.microsoft.com/en-us/azure/guidance/guidance-naming-conventions resource groups can only be 64 characters long. So a random 4 alphanumeric characters gives us over 1.6 million safe combinations, which should be more than enough. I wouldn't anticipate more than 10-20 outstanding resource groups in "deleting" status at a time and that would only be for things which are very slow to delete. This approach allows 58 characters to be reserved for the actual name portion (which is helpful if you use skip_teardown=True so you can examine its resources)